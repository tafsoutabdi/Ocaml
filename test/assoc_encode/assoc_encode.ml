open Tokenization.Assoc.Tokenizer
open Utils.Assoc
open Utils.Std

let print_context voc s = 
  Format.printf "voc = %s@.s = \"%s\"@.encode voc s = "
    (string_of_vocabulary voc) s

let () = 
  let voc = [("Hello, ", 13); ("world", 7); ("!", 12)] in
  let s = "Hello, world!!" in
  print_context voc s;
  print_id_list (encode voc s);
  Format.printf "@."

let () = 
let voc = [("world", 7); ("!", 12); ("Hello, ", 13)] in
let s = "Hello, world!!" in
print_context voc s;
print_id_list (encode voc s);
Format.printf "@."