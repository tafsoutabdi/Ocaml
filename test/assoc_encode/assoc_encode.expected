voc = [("Hello, ", 13); ("world", 7); ("!", 12)]
s = "Hello, world!!"
encode voc s = [13; 7; 12; 12]
voc = [("world", 7); ("!", 12); ("Hello, ", 13)]
s = "Hello, world!!"
encode voc s = [13; 7; 12; 12]
